src/ndt-client.js

```diff
@@ -98,6 +115,64 @@ NDTjs.prototype.checkEnvironmentSupport = function() {
 };
 
 /**
+ * A generic login creation system for NDT.
+ *
+ * @param {Number} desiredTests The types of tests requested from the server
+ *  signaled based on a bitwise operation of the test ids.
+ * @returns {Uint8Array} An array of bytes suitable for sending on a binary
+ *  websocket.
+ */
+
+NDTjs.prototype.makeNDTLogin = function(desiredTests) {
+  var messageBody = '{ "msg": "v3.5.5", "tests": "' +
+      String(desiredTests | 16) + '" }';
+
+  return this.makeNDTMessageArray(this.constants.messageType.MSG_EXTENDED_LOGIN,
+      messageBody);
+};
+
+/**
+ * A generic message creation system for NDT.
+ *
+ * @param {Number} messageType The type of message according to NDT's
+ *  specification.
+ * @param {String} messageContent The message to send the server.
+ * @returns {Uint8Array} An array of bytes suitable for sending on a binary
+ *  websocket.
+ */
+
+NDTjs.prototype.makeNDTMessage = function(messageType, messageContent) {
+   var messageBody = '{ "msg": "' + messageContent + '" }';
```

Comment:
Nice eye - that's an old enough hack (from last May or June, maybe?) that my eyes had glossed right over it.  Assuming that JSON works the same in all browsers and in node.js, this looks like a good cleanup to do.